homeassistant/components/electric_kiwi/coordinator.py

```diff
@@ -0,0 +1,81 @@
+"""Electric Kiwi coordinators."""
+from collections import OrderedDict
+from datetime import timedelta
+import logging
+
+import async_timeout
+from electrickiwi_api import ElectricKiwiApi
+from electrickiwi_api.exceptions import ApiException, AuthException
+from electrickiwi_api.model import Hop, HopIntervals
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+_LOGGER = logging.getLogger(__name__)
+
+HOP_SCAN_INTERVAL = timedelta(hours=2)
+
+
+class ElectricKiwiHOPDataCoordinator(DataUpdateCoordinator[Hop]):
+    """ElectricKiwi Data object."""
+
+    def __init__(self, hass: HomeAssistant, ek_api: ElectricKiwiApi) -> None:
+        """Initialize ElectricKiwiAccountDataCoordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            # Name of the data. For logging purposes.
+            name="Electric Kiwi HOP Data",
+            # Polling interval. Will only be polled if there are subscribers.
+            update_interval=HOP_SCAN_INTERVAL,
+        )
+        self._ek_api = ek_api
+        self.hop_intervals: HopIntervals = HopIntervals("", "", OrderedDict())
+
+    def get_hop_options(self) -> dict[str, int]:
+        """Get the hop interval options for selection."""
+        return {
+            f"{v.start_time} - {v.end_time}": k
+            for k, v in self.hop_intervals.intervals.items()
+        }
+
+    async def async_update_hop(self, hop_interval: int) -> Hop:
+        """Update selected hop and data."""
+        try:
+            self.async_set_updated_data(await self._ek_api.post_hop(hop_interval))
+        except AuthException as auth_err:
+            raise ConfigEntryAuthFailed from auth_err
+        except ApiException as api_err:
+            raise UpdateFailed(
+                f"Error communicating with EK API: {api_err}"
+            ) from api_err
+
+        return self.data
+
+    async def _async_update_data(self) -> Hop:
+        """Fetch data from API endpoint.
+
+        filters the intervals to remove ones that are not active
+        """
+        try:
+            # Note: asyncio.TimeoutError and aiohttp.ClientError are already
+            # handled by the data update coordinator.
+            async with async_timeout.timeout(60):
+                if len(self.hop_intervals.intervals) == 0:
```

Comment:
```suggestion
                if not len(self.hop_intervals.intervals):
```

Reply:
can't do according to the linter `not len()` so refactoring this type since its a dictionary.